src/index.ts

```diff
@@ -345,17 +355,125 @@ em.addListener('SANITIZED_CONFIG', function () {
   // TODO: Add function here
 })
 
-// EVENT EMITTER - RESPONSE_RECEIVED
-interface ResponseReceived {
+// EVENT EMITTER - PROBE_RESPONSE_RECEIVED
+interface ProbeResponseReceived {
   alerts: string[]
   response: AxiosResponseWithExtraData
 }
 
-// RESPONSE_RECEIVED - VALIDATE RESPONSE
-em.on(RESPONSE_RECEIVED, function (data: ResponseReceived) {
+// PROBE_RESPONSE_RECEIVED - VALIDATE RESPONSE
+em.on(PROBE_RESPONSE_RECEIVED, function (data: ProbeResponseReceived) {
   const res = validateResponse(data.alerts, data.response)
 
-  em.emit(RESPONSE_VALIDATED, res)
+  em.emit(PROBE_RESPONSE_VALIDATED, res)
+})
+
+// EVENT EMITTER - PROBE_STATUS_PROCESSED
+interface ProbeStatusProcessed {
+  probe: Probe
+  statuses?: StatusDetails[]
+  notifications?: Notification[]
+  validatedResponseStatuses: ValidateResponseStatus[]
+  totalRequests: number
+}
+
+interface ProbeSendNotification extends Omit<ProbeStatusProcessed, 'statuses'> {
+  index: number
+  status?: StatusDetails
+}
+
+interface ProbeSaveLogToDatabase
+  extends Omit<
+    ProbeStatusProcessed,
+    'statuses' | 'totalRequests' | 'validatedResponseStatuses'
+  > {
+  index: number
+  status?: StatusDetails
+}
+
+// PROBE_STATUS_PROCESSED
+em.on(PROBE_STATUS_PROCESSED, (data: ProbeStatusProcessed) => {
+  const {
+    probe,
+    statuses,
+    notifications,
+    totalRequests,
+    validatedResponseStatuses,
+  } = data
+
+  statuses
+    ?.filter((status) => status.shouldSendNotification)
+    ?.forEach((status, index) => {
+      em.emit(PROBE_SEND_NOTIFICATION, {
+        index,
+        probe,
+        status,
+        notifications,
+        totalRequests,
+        validatedResponseStatuses,
+      })
+
+      em.emit(PROBE_SAVE_LOG_TO_DATABASE, {
```

Comment:
You can just simply create sendNotification and saveLogToDatabase function, because I think there is no point to emit an event from here, at least for now.

Reply:
Nice, thanks!